hdf5-blosc2/src/blosc2_filter.c

```diff
@@ -367,6 +369,14 @@ size_t blosc2_filter_function(unsigned flags, size_t cd_nelmts,
 
     blosc2_storage storage = {.cparams=&cparams, .contiguous=false};
 
+    if (ndim > 1 && nbytes != chunksize) {
+      BLOSC_TRACE_INFO("Filter input size %lu does not match chunk data size %lu "
+                       "(e.g. Fletcher32 checksum added before compression step), "
+                       "using plain Blosc2 instead of B2ND",
+                       nbytes, chunksize);
+      ndim = -1;
```

Comment:
If I read correctly, this is still allowing to create compressed datasets with Fletcher32 being used prior to Blosc2, right?  I don't think choosing where Fletcher32 is used is supported by PyTables, but my guess is that this trace message is more for pure HDF5/h5py apps that can choose the order of the pipeline. Is that the intent?

Reply:
Yes, indeed it is. :slightly_smiling_face: I'd love to add a similar protection for the reading/decompression case, but unfortunately one can't say beforehand in the read buffer contains data that won't be used for decompression, and `blosc2_schunk_from_buffer` is ok with trailing bytes (which is fine IMO) and doesn't report whether it consumed the whole buffer or not. So the situation cannot be detected by the filter alone.

Thus in practice, following the optimized path for reading a B2ND schunk followed by a Fletcher32 checksum would work, but of course the checksum would not be verified at all, which doesn't feel like correct behaviour to me. That more aggressive optimization behaviour may be implemented (showing some warning or something at the PyTables level) but I'm not sure it's worth the effort.

Reply:
Added note on that in commit 9fee83a4.

Reply:
> Yes, indeed it is. 🙂 I'd love to add a similar protection for the reading/decompression case, but unfortunately one can't say beforehand in the read buffer contains data that won't be used for decompression, and `blosc2_schunk_from_buffer` is ok with trailing bytes (which is fine IMO) and doesn't report whether it consumed the whole buffer or not. So the situation cannot be detected by the filter alone.
> 
> Thus in practice, following the optimized path for reading a B2ND schunk followed by a Fletcher32 checksum would work, but of course the checksum would not be verified at all, which doesn't feel like correct behaviour to me. That more aggressive optimization behaviour may be implemented (showing some warning or something at the PyTables level) but I'm not sure it's worth the effort.

Ok, at this point I don't think checking the Fletcher32 checksum for the optimized path makes sense either; making it not crashing should be enough for now. In the long run, if such a functionality is desired, it would be better to implement a Fletcher32 (or equivalent) into the Blosc2 library, and call it from there.  Perhaps it would be nice to add a note about this in the docs.

Reply:
Yup, added the note in commit 6bbed526.